giraffe/src/components/geo/TrackMapLayer.tsx

```diff
@@ -0,0 +1,94 @@
+// Libraries
+import React from 'react'
+import {FunctionComponent} from 'react'
+import AntPath from 'react-leaflet-ant-path'
+import {CircleMarker} from 'react-leaflet'
+
+// Types
+import {GeoTable} from './processing/GeoTable'
+import {GeoTrackMapViewLayer} from '../../types/geo'
+import {Config} from '../../types'
+
+interface Props {
+  table: GeoTable
+  properties: GeoTrackMapViewLayer
+  stylingConfig: Partial<Config>
+}
+
+const DEFAULT_TRACK_COLOR = [{hex: '#FFC400'}, {hex: '#F90A13'}]
+const DEFAULT_TRACK_PALETTE = [
+  {hex: 'blue'},
+  {hex: 'red'},
+  {hex: 'green'},
+  {hex: 'brown'},
+  {hex: 'black'},
+  {hex: 'deeppink'},
+  {hex: 'olive'},
+]
+const DEFAULT_END_MARKER_RADIUS = 4
+
+export const TrackMapLayer: FunctionComponent<Props> = props => {
+  const {table, properties} = props
+  let colors = properties.colors || DEFAULT_TRACK_COLOR
+  if (!properties.colors && properties.randomColors)
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.